pkg/target/target_builder.go

```diff
@@ -63,14 +63,14 @@ func (b *targetBuilderImpl) Init(t Target) TargetBuilder {
 	return b
 }
 
-func (b *targetBuilderImpl) SetGarden(name string) TargetBuilder {
+func (b *targetBuilderImpl) SetGarden(shortOrIdentity string) TargetBuilder {
 	b.actions = append(b.actions, func(t *targetImpl) error {
-		gardenName, err := b.config.GardenName(name)
+		garden, err := b.config.Garden(shortOrIdentity)
```

Comment:
possible nil pointer dereference. Check if config is nil in `NewTargetBuilder` and return the error

Reply:
ok